libraries/Netdump/src/NetdumpPacket.cpp

```diff
@@ -0,0 +1,381 @@
+        case 128 : sstr.printf_P(PSTR("ping request")); break;
+        case 135 : sstr.printf_P(PSTR("Neighbour solicitation")); break;
+        case 136 : sstr.printf_P(PSTR("Neighbour advertisement")); break;
+        default: sstr.printf_P(PSTR("type(0x%02x)"), getIcmpType()); break;
+        }
+    }
+    sstr.printf("\r\n");
+}
+
+void Packet::IGMPtoString(PacketDetail netdumpDetail, StreamString& sstr) const
+{
+    switch (getIgmpType())
+    {
+    case 1 : sstr.printf_P(PSTR("Create Group Request")); break;
+    case 2 : sstr.printf_P(PSTR("Create Group Reply")); break;
+    case 3 : sstr.printf_P(PSTR("Join Group Request")); break;
+    case 4 : sstr.printf_P(PSTR("Join Group Reply")); break;
+    case 5 : sstr.printf_P(PSTR("Leave Group Request")); break;
+    case 6 : sstr.printf_P(PSTR("Leave Group Reply")); break;
+    case 7 : sstr.printf_P(PSTR("Confirm Group Request")); break;
+    case 8 : sstr.printf_P(PSTR("Confirm Group Reply")); break;
+    case 0x11 : sstr.printf_P(PSTR("Group Membership Query")); break;
+    case 0x12 : sstr.printf_P(PSTR("IGMPv1 Membership Report")); break;
+    case 0x22 : sstr.printf_P(PSTR("IGMPv3 Membership Report")); break;
+    default: sstr.printf_P(PSTR("type(0x%02x)"), getIgmpType()); break;
+    }
+    sstr.printf_P(PSTR("\r\n"));
+}
+
+void Packet::IPtoString(PacketDetail netdumpDetail, StreamString& sstr) const
+{
+    sstr.printf_P(PSTR("%s>%s "), sourceIP().toString().c_str(), destIP().toString().c_str());
+    sstr.printf_P(PSTR("Unknown IP type : %d\r\n"), ipType());
+    printDetail(sstr, PSTR("           H "), &data[ETH_HDR_LEN], getIpHdrLen(), netdumpDetail);
+    printDetail(sstr, PSTR("           D "), &data[ETH_HDR_LEN + getIpHdrLen()], getIpTotalLen() - getIpHdrLen(), netdumpDetail);
+}
+
+void Packet::UKNWtoString(PacketDetail netdumpDetail, StreamString& sstr) const
+{
+    sstr.printf_P(PSTR("Unknown EtherType 0x%04x Src : "), ethType());
+    MACtoString(0, sstr);
+    sstr.printf_P(PSTR(" Dst : "));
+    MACtoString(6, sstr);
+    sstr.printf_P(PSTR("\r\n"));
+}
+
+const String Packet::toString() const
+{
+    return toString(PacketDetail::NONE);
+}
+
+
+const String Packet::toString(PacketDetail netdumpDetail) const
+{
+    StreamString sstr;
+    sstr.reserve(128);
+
+    sstr.printf_P(PSTR("%d %3s %-4s "), netif_idx, out ? "out" : "in ", packetType().toString().c_str());
+
+    if (netdumpDetail == PacketDetail::RAW)
+    {
+        sstr.printf_P(PSTR(" : "));
+        for (auto at : thisAllPacketTypes)
+        {
+            sstr.printf_P(PSTR("%s "), at.toString().c_str());
+        }
+        sstr.printf_P(PSTR("\r\n"));
+        printDetail(sstr, PSTR("           D "), data, packetLength, netdumpDetail);
+        return sstr;
+    }
+
+    switch (thisPacketType)
+    {
+    case PacketType::ARP :
+    {
+        ARPtoString(netdumpDetail, sstr);
+        break;
+    }
+    case PacketType::MDNS :
+    case PacketType::DNS :
```

Comment:
It is needed, for the upcoming c++17's g++9/10 - (or is it just from c/c++ gcc7+?) - to add 
```
// falls through
```
(or `/* fall through */`, `/* Fall through */`, ...)
in everyplace where one intentionnally doesn't add a `break`.